source/patterns/@aws-solutions-constructs/core/lib/cloudfront-distribution-helper.ts

```diff
@@ -23,6 +23,7 @@ import {
 } from './cloudfront-distribution-defaults';
 import { overrideProps, addCfnSuppressRules } from './utils';
 import { createLoggingBucket } from './s3-bucket-helper';
+import { Construct } from '@aws-cdk/core';
```

Comment:
So why are we doing this? There are still cdk. references to other names, so now we have 2 different ways to reference names in core. That doesn't seem like an improvement.

Reply:
Short answer, yes it is not an improvement by any means, it is to keep the same codebase that will build for both CDK v1 and CDK v2
Long answer, In CDK v2 'Construct' is migrated from '@aws-cdk/core' package to 'constructs' package in NPM. If we separate out the import statements for  'Construct', then during the CDK v2 build it will be replaced from:

```
import { Construct } from '@aws-cdk/core';
```
to
```
import { Construct } from 'constructs';
```

Other core classes e.g. `App`, `Aws`,`Duration`, etc. are migrated from `@aws-cdk/core` to `aws-cdk-lib`

Yep, it's a mess!!!